subsys/bluetooth/controller/hci/hci.c

```diff
@@ -1719,6 +1719,17 @@ static void le_set_scan_param(struct net_buf *buf, struct net_buf **evt)
 				    cmd->addr_type, cmd->filter_policy);
 
 	*evt = cmd_complete_status(status);
+
+	if (IS_ENABLED(CONFIG_BT_CTLR_ADV_EXT) &&
```

Comment:
From commit message I understand that this code is required only after HCI reset is executed.
Can you elaborate why it is done here instead of reset context?

Reply:
Good catch, I will move this to reset context.